### PR TITLE
feat(condo): INFRA-764 reset user rate-limit

### PR DESCRIPTION
**Pull request by SavelevMatthew**
## TODO:

- [x] Generate schemaDocs from resetters
- [x] Write good tests to test rate-limit (Ok, good tests are not possible, since we'll probably disable rate-limit in CI to avoid unauthed errors, so I'll just set variable and reset it like @nomerdvadcatpyat did with auth)

### CODE REVIEW

File: apps/condo/domains/user/utils/limits.js [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+        const identifierType = getIdentifierType(identifier)
+        if (!this.supportedIdentifiers.includes(identifierType)) {
+            return false
+        }
+
+        return await this.guard.deleteCounter(`${AUTH_COUNTER_LIMIT_TYPE}:${identifier}`)
+    }
+}
+
+class RateLimitResetter {
```
Comment (nomerdvadcatpyat) [inline apps/condo/domains/user/utils/limits.js:35]: I think it's better to move Resetters to separate files

File: apps/condo/domains/user/utils/limits.js [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+    constructor () {
+        this.guard = new RedisGuard()
+    }
+
+    async checkExistence (identifier) {
+        const identifierType = getIdentifierType(identifier)
+        if (!this.supportedIdentifiers.includes(identifierType)) {
+            return false
+        }
```
Comment (nomerdvadcatpyat) [inline apps/condo/domains/user/utils/limits.js:20]: This logic can be a separate method, for example, `isIdentifierSupported`.
And reused here, in reset method and in ResetUserLimitAction.afterChange

File: apps/condo/domains/user/utils/identifiers.spec.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
+const { faker } = require('@faker-js/faker')
+
+const { createTestPhone } = require('@condo/domains/user/utils/testSchema')
+
+const { getIdentifierType, IPv4_TYPE, UUID_TYPE, PHONE_TYPE } = require('./identifiers')
+
+describe('User identifiers utils', () => {
+    describe('getIdentifierType', () => {
+        const SAMPLES_COUNT = 1000
```
[the remark this answers is not in the document]
Comment (SavelevMatthew) [reply apps/condo/domains/user/utils/identifiers.spec.js:9]: Yep, all tests are done under a second locally

File: apps/condo/domains/user/utils/limits.js (hunk 1 above)
Comment (SavelevMatthew) [reply apps/condo/domains/user/utils/limits.js:35]: Separated logic into `utils/limits/guards`

File: apps/condo/domains/user/utils/limits.js (hunk 2 above)
Comment (SavelevMatthew) [reply apps/condo/domains/user/utils/limits.js]: Moved this logic into `AbstractResetter`

**Review by nomerdvadcatpyat — APPROVED** [review]

**Pull request merged**